Sources/AblyAssetTrackingCore/ResultHandler.swift

```diff
@@ -1,10 +1,68 @@
-/**
- Completion handler for operations
- */
-public typealias ResultHandler<T: Any> = (Result<T, ErrorInformation>) -> Void
+import Foundation
 
-public extension Result where Success == Void {
-    static var success: Result {
+public typealias ResultHandler = (ATResult) -> Void
+
+@objc
+public protocol Resultable {
+    var success: Any? { get }
+    var failure: ErrorInformation? { get }
+}
+
+@objcMembers
+public class ATResult: NSObject, Resultable {
```

Comment:
Following ably-cocoa naming convention, shouldn't it be AATResult?
```
@objc(AATResult)
public class Result: NSObject, Resultable {
```

Reply:
I agree the prefix should be `AAT` but I don't want to change the name to `Result` because it'll interfere with Swift `Result<T,U>`

Reply:
ac61e71875421b980ba7438c6a351bcf0402db1d